dataGeneration/generate-cosine-data-multi-entity.py

```diff
@@ -0,0 +1,296 @@
+parser.add_argument("-np", "--number-of-process", type=int, default=1000, help="number of 'process' entities, deafult is set to 1000, there will be two keyword categories in this index (must be at least 1)" )
+parser.add_argument("-hd", "--number-of-historical-days", type=int, default=2, help="number of day of historical data to ingest, defaults to 2")
+parser.add_argument("-u", "--username", type=str, default="admin", help="username for authentication if security is true")
+parser.add_argument("-pass", "--password", type=str, default="admin", help="password for authentication if security is true")
+
+args = parser.parse_args()
+
+URL = args.endpoint
+SECURITY = args.security
+INDEX_NAME = args.index_name
+SHARD_NUMBER = args.shards
+THREADS = args.threads
+
+#deafult numbers of 1000 host and 1000 process mean a total of 1 million entities
+HOST_NUMBER = args.number_of_host
+PROCESS_NUMBER = args.number_of_process
+
+#default of 1008 points with ingestion frequency set to 600 means there will basically be 1008 intervals = 7 days * 144 intervals/day
+POINTS = args.points 
+INGESTION_FREQUENCY = args.ingestion_frequency
+BULK_SIZE = args.bulk_size
+USERNAME = args.username
+PASSWORD = args.password
+NUMBER_OF_HISTORICAL_DAYS = args.number_of_historical_days
+
+
+index_name = "_index"
+timestamp_name = "@timestamp"
+cpu_name = "cpuTime"
+mem_name = "jvmGcTime"
+host_name = "host"
+host_prefix = "host"
+process_name = "process"
+process_prefix = "process"
+client = []
+
+'''
+    Generate index INDEX_NAME
+'''
+def create_index(es, INDEX_NAME, shard_number):
+    # First, delete the index if it exists
+    print("Deleting index if it exists...")
+    es.indices.delete(index=INDEX_NAME, ignore=[400, 404])
+
+    # Next, create the index
+    print("Creating index \"{}\"...".format(INDEX_NAME))
+    request_body = {
+        "settings":{
+          "number_of_shards":shard_number,
+          "number_of_replicas": 0, # increase this number after indexing
+          "translog.durability":"async", # default: request
+          "refresh_interval":-1, # default: 1, remember to change this after finishing indexing process or just _refresh once at least if index wont be changed again
+       },
+       "mappings":{
+          "properties":{
+             "@timestamp":{
+                "type":"date"
+             },
+             "cpuTime":{
+                "type":"double"
+             },
+             "jvmGcTime":{
+                "type":"double"
+             },
+             "host":{
+                "type":"keyword"
+             },
+             "process":{
+                "type":"keyword"
+             }
+          }
+       }
+    }
+
+    es.indices.create(index=INDEX_NAME, body=request_body)
+
+
+'''
+    Posts a document(s) to the index
+'''
+@retry(delay=2)
```

Comment:
the default configuration retries using exponential backoff. Is is better than a hardcoded 2 seconds?

Reply:
this retry package default is at 0 https://pypi.org/project/retry/. But I can add a backoff param set to 2